module-3/app/service/client/mysfitsTableClient.go

```diff
@@ -0,0 +1,188 @@
+	Name            string `json:"Name"`
+	Species         string `json:"Species"`
+	Description     string `json:"Description"`
+	Age             int    `json:"Age"`
+	GoodEvil        string `json:"GoodEvil"`
+	LawChaos        string `json:"LawChaos"`
+	ThumbImageUri   string `json:"ThumbImageUri"`
+	ProfileImageUri string `json:"ProfileImageUri"`
+	Likes           int    `json:"Likes"`
+	Adopted         bool   `json:"Adopted"`
+}
+
+// Items stores a list of Items
+type Items []Item
+
+// Get items as array of structs
+func getItems(items []map[string]*dynamodb.AttributeValue) Items {
+	var mysfitList Items
+
+	err := dynamodbattribute.UnmarshalListOfMaps(items, &mysfitList)
+	if err != nil {
+		println("Got error unmarshalling items:")
+		println(err.Error())
+		return nil
+	}
+
+	return mysfitList
+}
+
+// getJSONStringFromItems creates a JSON string from the items from a scan or query
+func getJSONStringFromItems(items []map[string]*dynamodb.AttributeValue) string {
+	output := "{\"mysfits\": ["
+	myItems := getItems(items)
+
+	len := len(myItems)
+
+	for i, item := range myItems {
+		iJSON, err := json.Marshal(item)
+		if err != nil {
+			fmt.Println(err)
+			return ""
+		}
+
+		if i == len-1 {
+			output += string(iJSON)
+		} else {
+			output += string(iJSON) + ","
+		}
+	}
+
+	return output + "]}"
+}
+
+// GetAllMysfits gets all table items
+func GetAllMysfits() string {
+	// Create a DynamoDB client using our default credentials and region.
+	sess := session.Must(session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	}))
+
+	// Create DynamoDB client
+	svc := dynamodb.New(sess)
+
+	// Retrieve all Mysfits from DynamoDB using the DynamoDB scan operation.
+	// Note: The scan API can be expensive in terms of latency when a DynamoDB
+	// table contains a high number of records and filters are applied to the
+	// operation that require a large amount of data to be scanned in the table
+	// before a response is returned by DynamoDB. For high-volume tables that
+	// receive many requests, it is common to store the result of frequent/common
+	// scan operations in an in-memory cache. DynamoDB Accelerator (DAX) or
+	// use of ElastiCache can provide these benefits. But, because out Mythical
+	// Mysfits API is low traffic and the table is very small, the scan operation
+	// will suit our needs for this workshop.
+
+	input := &dynamodb.ScanInput{
+		TableName: aws.String("MysfitsTable"),
+	}
+
+	result, err := svc.Scan(input)
+	if err != nil {
+		Info.Print("Got error scanning table:")
+		return ""
+	}
+
+	// loop through the returned mysfits and add their attributes to a new map
+	// that matches the JSON response structure expected by the frontend.
+	Init(os.Stdout)
+	Info.Print(result.Items)
+
+	jsonValue := getJSONStringFromItems(result.Items)
+
+	return jsonValue
+}
+
+// QueryMysfits gets only the specified items
+func QueryMysfits(filter string, value string) string {
+	Init(os.Stdout)
```

Comment:
See previous comment about the use of the Init method

Reply:
I've changed init to both set the output path (stderr in the client for now) and the output format to the web page (HTML for now). I only call it in the service client.